File: apps/web/src/features/tasks/status.ts
import type { TaskStatus } from '@tooday/shared';

/** 상태 표시 순서 — 보드 세그먼트·상태 선택 시트에서 공용 */
export const STATUS_ORDER: readonly TaskStatus[] = ['todo', 'doing', 'done'] as const;

/** 상태 → Dot tone (점 색) */
export const STATUS_DOT_TONE: Record<TaskStatus, 'neutral' | 'primary' | 'success'> = {
  todo: 'neutral',
  doing: 'primary',
  done: 'success',
};

/** 상태 → Chip tone (알약 색) */
export const STATUS_CHIP_TONE: Record<TaskStatus, 'neutral' | 'brand' | 'success'> = {
  todo: 'neutral',
  doing: 'brand',
  done: 'success',
};

File: rust/crates/web/src/entities/task/status.rs
use tooday_shared::TaskStatus;

use crate::shared::ui::{ChipTone, DotTone};

/// 상태 표시 순서 — 보드 세그먼트(projects)·상태 선택 시트(tasks) 등 여러 feature가 공용한다
pub const STATUS_ORDER: [TaskStatus; 3] = [TaskStatus::Todo, TaskStatus::Doing, TaskStatus::Done];

/// 상태 → Dot tone (점 색)
pub fn status_dot_tone(status: TaskStatus) -> DotTone {
    match status {
        TaskStatus::Todo => DotTone::Neutral,
        TaskStatus::Doing => DotTone::Primary,
        TaskStatus::Done => DotTone::Success,
    }
}

/// 상태 → Chip tone (알약 색)
pub fn status_chip_tone(status: TaskStatus) -> ChipTone {
    match status {
        TaskStatus::Todo => ChipTone::Neutral,
        TaskStatus::Doing => ChipTone::Brand,
        TaskStatus::Done => ChipTone::Success,
    }
}
